### PR TITLE
[Python] New page on ResultSummary

### PR DESCRIPTION
**Pull request by stefano-ottolenghi**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-drivers-393.surge.sh

File: python-manual/modules/ROOT/pages/result-summary.adoc [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# Explore the query execution summary
+
+After all results coming from a query have been processed, the server ends the transaction by returning a summary of execution.
+It comes as a link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#resultsummary[`ResultSummary`] object, and it contains information among which:
+
+- **Query counters** -- What changes the query triggered on the server
+- **Notifications** -- Extra information raised by the server while running the query
+- **Query plan** -- How the database executed the query
```
Comment (robsdedude) [inline python-manual/modules/ROOT/pages/result-summary.adoc]: Is the plan more mention worthy than the profile?

Comment (robsdedude) [inline python-manual/modules/ROOT/pages/result-summary.adoc]: Does it make sense to mention that the plan will be empty unless you send a `PLAN ...` query? (At least I think it is)

File: python-manual/modules/ROOT/pages/result-summary.adoc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+----
+records, result_summary, keys = driver.execute_query("""
+    UNWIND ["Alice", "Bob"] AS name
+    MERGE (p:Person {name: name})
+    """, database_="neo4j",
+)
+----
+
+If you are using xref:transactions.adoc[transaction functions], or a xref:transformers.adoc#_custom_transformers[custom transformer] with `Driver.execute_query()`, you can retrieve the query execution summary with the method `Result.consume()`.
+**Notice that once you ask for the execution summary, the transaction is closed and the result stream exhausted**.
```
Comment (robsdedude) [inline python-manual/modules/ROOT/pages/result-summary.adoc]: It does not close the transaction. `commit` or `rollback` does.

File: python-manual/modules/ROOT/pages/result-summary.adoc [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    raw_category='PERFORMANCE',
+    position=SummaryNotificationPosition(line=1, column=22, offset=21)
+)]
+"""
+----
+
+
+### Filtering notifications
+
+By default, the server analyses each query for all categories and severity of notifications.
```
Comment (robsdedude) [inline python-manual/modules/ROOT/pages/result-summary.adoc:167]: Ooh. I've always been wondering if we prefer our docs in uk-en or us-en.

File: python-manual/modules/ROOT/pages/result-summary.adoc [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+If you are only interested in some of them, or if you want to disable them altogether, you can use the driver parameters `notifications_min_severity` and/or `notifications_disabled_categories` when creating a `Driver` instance in the first place (available from version 5.7).
```
Comment (robsdedude) [inline python-manual/modules/ROOT/pages/result-summary.adoc]: This config setting also exists on session level. Not sure if worth mentioning.

**Review by robsdedude — APPROVED** [review]
:clap:

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 1 above)
Comment (stefano-ottolenghi) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: You mean more worth than notifications? In case, yes I agree. I swapped the order

Comment (stefano-ottolenghi) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: It says `If you prefix your query with the keyword PROFILE` at the start of the query plan section, I think more becomes too much.

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 3 above)
Comment (stefano-ottolenghi) [reply python-manual/modules/ROOT/pages/result-summary.adoc:167]: Yeah. I'll pretend I picked one _on purpose_.

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 4 above)
Comment (stefano-ottolenghi) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: Ah good to know! I assume values for those parameters at the session level would override the same parameters given when creating the driver, right?

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 1 above)
Comment (robsdedude) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: I mean it only speaks about plans. Profiles are left out completely. I mean they are very similar, but not quite the same thing. `PLAN ...` vs. `EXPLAIN ...` queries.

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 4 above)
Comment (robsdedude) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: Right. It's session config overwrites driver config overwrites server default.

File: python-manual/modules/ROOT/pages/result-summary.adoc (hunk 1 above)
Comment (robsdedude) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: > How the database executed the query

I think that's what the profile is

> How the database would execute the query

is more like what the plan is.

Comment (stefano-ottolenghi) [reply python-manual/modules/ROOT/pages/result-summary.adoc]: I added `EXPLAIN` info and flipped the exposition. Checked with cypher teams for proper wording.

**Pull request merged**